include/deal.II/numerics/history.h

```diff
@@ -0,0 +1,248 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_storage_h
+#define dealii_storage_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/std_cxx14/memory.h>
+
+#include <deque>
+#include <type_traits>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A helper class to store a vector of unique pointers to objects of type `T`.
```

Comment:
Can you rewrite this somehow in a way so that it (i) collection, not just helper-class, (ii) makes it clear that it's a *finite-size* collection, and (iii) omits the *implementation detail* that it stores unique pointers.